src/main/java/seedu/address/commons/core/index/Index.java

```diff
@@ -37,14 +37,21 @@ public int getOneBased() {
      * Creates a new {@code Index} using a zero-based index.
      */
     public static Index fromZeroBased(int zeroBasedIndex) {
-        return new Index(zeroBasedIndex);
+        return new Index(zeroBasedIndex, true);
     }
 
     /**
      * Creates a new {@code Index} using a one-based index.
      */
     public static Index fromOneBased(int oneBasedIndex) {
-        return new Index(oneBasedIndex - 1);
+        return new Index(oneBasedIndex - 1, true);
+    }
+
+    /**
+     * Creates a new {@code Index} using a zero-based index, bypassing non-positive checks (only used for delete-task).
+     */
+    public static Index oneBasedNoConstraints(int index) {
+        return new Index(index - 1, false);
```

Comment:
Just wondering, may I know the reasoning these changes?  For the oneBasedNoConstraints the javadocs seem to suggest non-positive checks is bypassed for delete-task which I don't think is right? Just wanted to get a clearer understanding.

Reply:
agree with @DanzaSeah why are non-positive checks bypassed for delete task?

Reply:
This bypass is because right now the error message for indexes that are <=0 and error msg for indexes >max index are different

And instead of just throwing an error, I wanted to put the range of valid index for users, and only model has the valid range so it has to fall through to the `DeleteTaskCommand::execute` which will be caught (this "catching at execute" was initially only the case for > max index so now <=0 is also caught there)